integration/query_projection_compat_test.go

```diff
@@ -55,6 +55,10 @@ func TestQueryProjectionCompat(t *testing.T) {
 	providers := append(shareddata.AllProviders(), topLevelFieldsIntegers)
 
 	testCases := map[string]queryCompatTestCase{
+		"EmptyProjection": {
```

Comment:
Can we also add something like

```
		"NilProjection": {
			filter:     bson.D{},
			projection: nil,
		},
```

or it's not possible for the same test?

Reply:
Thanks for the suggestion, added this test case.